src/lib.rs

```diff
@@ -0,0 +1,555 @@
+                // Invalidate line data: Skip
+                if attr.len() != 2 {
+                    continue;
+                }
+                attrs.insert(attr[0].to_string(), attr[1].parse().unwrap());
+            }
+
+            // Update edge set
+            if !directed {
+                update_edge_set(&mut edges, target, source, attrs.clone())
+            }
+            update_edge_set(&mut edges, source, target, attrs);
+
+            if no_node_file {
+                if !vertices.contains_key(&source) {
+                    vertices.insert(source, Vertex::new(source));
+                }
+                if !vertices.contains_key(&target) {
+                    vertices.insert(target, Vertex::new(target));
+                }
+            }
+        }
+
+        //Read vertices if node_file not `None`
+        if !no_node_file {
+            let path = self.node_file.clone().unwrap();
+            let node_file = File::open(path).expect("Please check `node_file` path");
+
+            let buf_reader = BufReader::new(node_file);
+            for origin_line in buf_reader.lines() {
+                // Split file data into structure
+                if let Err(_e) = origin_line {
+                    continue;
+                }
+
+                let line = origin_line.unwrap();
+                let edge_info: Vec<&str> = line.split(&self.separate).collect();
+                let id: T = edge_info[0].parse().unwrap();
+
+                // Generate node
+                let mut attrs: HashMap<String, Attr> = HashMap::new();
+                for i in 1..edge_info.len() {
+                    let attr: Vec<&str> = edge_info[i].split("=").collect();
+                    // Invalidate line data: Skip
+                    if attr.len() != 2 {
+                        continue;
+                    }
+                    attrs.insert(attr[0].to_string(), attr[1].parse().unwrap());
+                }
+                vertices.insert(id, Vertex { id, attrs });
+            }
+        }
+
+        let mut g = Graph {
+            graph_type: self.graph_type.clone(),
+            schema: self.schema.clone(),
+            node_file: self.node_file.clone(),
+            edge_file: self.edge_file.clone(),
+            separate: self.separate.clone(),
+            vertices: vec![],
+            edges: vec![],
+            neighbour_index: HashMap::new(),
+        };
+
+        // Generate edge index
+        for (id, v) in vertices {
+            let edge_opt = edges.get_mut(&id);
+            if let None = edge_opt {
+                continue;
+            }
+            let mut edge_vec = edge_opt.unwrap();
+            g.neighbour_index.insert(
+                id,
+                (
+                    self.neighbour_index.len(),
+                    edge_vec.len() + self.neighbour_index.len(),
+                ),
+            );
+            g.edges.append(&mut edge_vec);
+            g.vertices.push(v);
+        }
+        return g;
```

Comment:
If you want to return something in the last line, simply write
`g` instead of `return g`